### PR TITLE
MDEV-15408 Confusing error message upon ER_VERS_FIELD_WRONG_TYPE while omitting UNSIGNED in BIGINT

### PR DESCRIPTION
**Pull request by kevgs**
Improve diagnostics a little bit.

[Buildbot](http://buildbot.askmonty.org/buildbot/grid?category=main&branch=tt-10.3-mdev-15408)

I'm contributing this new code of the whole pull request, including one or several files that are either new files or modified ones, under the BSD-new license.

### CODE REVIEW

File: sql/handler.cc [hunk 1]
```diff
@@ -7218,78 +7218,102 @@ bool Vers_parse_info::check_conditions(const LString &table_name,
...
-      error:
-        my_error(ER_VERS_FIELD_WRONG_TYPE, MYF(0), f->field_name.str,
-                 check_unit == VERS_TIMESTAMP ?
-                 "TIMESTAMP(6)" :
-                 "BIGINT(20) UNSIGNED",
-                 table_name.str);
-        return true;
-      }
-      check_unit= f_check_unit;
+      require_timestamp(row_start_name, table_name);
```
Comment (vuvova) [inline sql/handler.cc]: it looks like if one writes `create table t1 (...row_start BIGINT) engine=myisam` you'll complain "need BIGINT UNISIGNED" and after user has done that you'll change your mind and complain "need TIMESTAMP(6)"

Better to test `!native` first, don't you think?

File: sql/handler.cc [hunk 2]
```diff
@@ -7218,78 +7218,102 @@ bool Vers_parse_info::check_conditions(const LString &table_name,
...
+  else
+  {
+    require_timestamp(row_start_name, table_name);
+    return true;
   }
 
-  my_error(ER_MISSING, MYF(0), table_name.str, found_flag & VERS_SYS_START_FLAG ?
-           "ROW END" : found_flag ? "ROW START" : "ROW START/END");
-  return true;
+  if (check_unit == VERS_TRX_ID && !TR_table::use_transaction_registry)
```
Comment (vuvova) [inline sql/handler.cc]: you don't do ER_MISSING anymore? where is it tested now?

File: sql/handler.cc [hunk 3]
```diff
@@ -7218,78 +7218,102 @@ bool Vers_parse_info::check_conditions(const LString &table_name,
...
-  while (Create_field *f= it++)
+  const Create_field *row_start= NULL;
+  const Create_field *row_end= NULL;
+  while (const Create_field *f= it++)
   {
-    vers_sys_type_t f_check_unit= VERS_UNDEFINED;
-    uint sys_flag= f->flags & VERS_SYSTEM_FIELD;
+    if (f->flags & VERS_SYS_START_FLAG && !row_start)
+      row_start= f;
+    if (f->flags & VERS_SYS_END_FLAG && !row_end)
```
[the remark this answers is not in the document]
Comment (kevgs) [reply sql/handler.cc]: It is checked in parser in `LEX::last_field_generated_always_as_row_start_or_end()`.
So, this code is not needed.

File: sql/handler.cc (hunk 2 above)
Comment (kevgs) [reply sql/handler.cc]: It is checked in `Vers_parse_info::check_conditions()` so this code is not needed here.

File: sql/handler.cc (hunk 1 above)
Comment (kevgs) [reply sql/handler.cc]: Yes, it was possible that MariaDB will suggest `bigint` for MyISAM. Fixed now.

File: sql/handler.cc (hunk 3 above)
[the remark this answers is not in the document]
Comment (vuvova) [reply sql/handler.cc]: Is there a test case for that? If yes, good nothing to do. If no — please add, in a separate commit.

**Comment by kevgs** [comment]
It's not good that `Vers_parse_info::check_sys_fields()` doesn't perform comprehensive check for partitions table. But that was the case before this patch and it just makes it more obvious.

For partitions table `handler::vers_can_native()` is used. And it can't be used too early after parsing.

File: sql/handler.cc (hunk 3 above)
[the remark this answers is not in the document]
Comment (kevgs) [reply sql/handler.cc]: It exists. When system versioning table is created as `CREATE ... LIKE` there could be 'duplicated' row start/end fields. And we interested only in first `row_start` and `row_end` occurrences.

**Pull request closed without merging**